sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_settings_client.py

```diff
@@ -72,6 +72,17 @@ def update_setting(self, setting: KeyVaultSetting, **kwargs) -> KeyVaultSetting:
         :rtype: ~azure.keyvault.administration.KeyVaultSetting
         :raises: :class:`~azure.core.exceptions.HttpResponseError`
         """
+        # Challenge authentication doesn't work as expected with this method's request endpoint
+        # If we need to prompt an auth challenge, do so using `get_setting` instead
+        # (Note: we prompt a challenge instead of fetching a token directly so the token has correct permissions)
+        challenge_cached = HttpChallengeCache.get_challenge_for_url(self._vault_url)
+        if not challenge_cached:
+            # `get_setting` is just used to prompt a challenge; we don't want to stop on any errors it may raise
+            try:
+                self._client.get_setting(vault_base_url=self._vault_url, setting_name=setting.name, **kwargs)
+            except Exception:  # pylint:disable=broad-except
```

Comment:
Even it is a fail to connect error?

Reply:
Yes; my thinking was that errors related to configuration (like connection errors) would be raised again anyway on the PATCH request. I was thinking that raising an error caused by `get_setting` when the user called `update_setting` could be confusing, so I was hoping to only raise for errors on the PATCH request.

That being said, the failing `get_setting` request would still be raised in the context of the `update_setting` method in a stack trace, so I could be convinced to remove this `try/except`

Reply:
Yeah, after thinking about this a bit more, I'm going to remove the error ignoring behavior. Some exceptions that would be raised by `get_setting` are ones we'd want to see, like authentication failures. The stack trace probably wouldn't be overly confusing to most users, and there are some errors that would be silly to trigger twice for the sake of failing on a PATCH instead of a GET.

Reply:
Take a step back.

Why we want to play the patch in client library while this is a service bug? Shouldn't service fix it?

Reply:
The service is fixing it, but it takes a good while for KV service fixes to deploy. If we don't provide the CLI team with a workaround, they'll likely implement one of their own. In the meantime, customers would also be able to hit this scenario, though it's unlikely -- and it's pretty tricky to figure out

Reply:
It's more likely that CLI users will hit this bug because the CLI uses short-lived clients. Starting a session to update a setting value the user is aware of is a much more likely scenario, and the team was hoping for an SDK-side workaround that they can consume this week in order to meet their release goal for this sprint and add support for this feature

Reply:
Do we have ETA when the fix will be deployed?

Reply:
As of yesterday I heard from the service team that they were creating a bug fix. Historically, once a fix is ready it takes some time to become available to end users -- on the scale of weeks, if I recall correctly.

Reply:
If the entire PR is to work around the issue, I would prefer to push it into CLI code.

We don't want customers to keep paying the cost for a temporary workaround.